### PR TITLE
Feature/check autogen

### PR DESCRIPTION
**Pull request by stmuk**
Main purpose is to alert via CI to swagger.yaml being out of sync with api/api.go etc. This is currently the case so the CI should fail with this merge!

Also add an extended make cover target with a summary and html support.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -121,8 +127,20 @@ update-schema:
...
+
+	@go generate cmd/dp-find-insights-poc-api/generate.go
+
+#
+# check no manual commits to auto-generated files on a clean tree
+#
+
+.PHONY: check-generate
+check-generate: generate
+	  @[ "$$(git diff)" = "" ] || (echo "commits to autogen file?" && exit 1)
```
Comment (VivianAllen) [inline Makefile:145]: could the error message for this be made any more specific? (which file, etc)

File: api/api.go [hunk 2]
```diff
@@ -26,7 +26,7 @@ type GetDevHelloDatasetParams struct {
 	Location    *string   `json:"location,omitempty"`
 	Radius      *int      `json:"radius,omitempty"`
 	Polygon     *string   `json:"polygon,omitempty"`
-	Censustable *string   `json:"polygon,omitempty"`
+	Censustable *string   `json:"censustable,omitempty"`
```
Comment (VivianAllen) [inline api/api.go:29]: this might have been me (sorry if so) - thanks for fixing

File: Makefile (hunk 1 above)
Comment (VivianAllen) [reply Makefile:145]: discussed offline, OOS

**Review by VivianAllen — APPROVED** [review]
really nice! ✅

File: Makefile (hunk 1 above)
Comment (stmuk) [reply Makefile:145]: https://github.com/ONSdigital/dp-find-insights-poc-api/issues/124

**Pull request merged**